dspace-api/src/main/java/org/dspace/statistics/export/processor/ExportEventProcessor.java

```diff
@@ -0,0 +1,297 @@
+            .append(URLEncoder.encode(getCurrentDateString(), UTF_8));
+
+        return data.toString();
+    }
+
+    /**
+     * Get the current date
+     *
+     * @return the current date as a string
+     */
+    protected String getCurrentDateString() {
+        return new DCDate(new Date()).toString();
+    }
+
+    /**
+     * Checks if an item should be processed
+     *
+     * @param item to be checked
+     * @return whether the item should be processed
+     * @throws SQLException
+     */
+    protected boolean shouldProcessItem(Item item) throws SQLException {
+        if (item == null) {
+            return false;
+        }
+        if (!item.isArchived()) {
+            return false;
+        }
+        if (itemService.canEdit(context, item)) {
+            return false;
+        }
+        if (!shouldProcessItemType(item)) {
+            return false;
+        }
+        if (!shouldProcessEntityType(item)) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Checks if the item's entity type should be processed
+     *
+     * @param item to be checked
+     * @return whether the item should be processed
+     * @throws SQLException
+     */
+    protected boolean shouldProcessEntityType(Item item) throws SQLException {
+        Entity entity = entityService.findByItemId(context, item.getID());
+        EntityType type = entityService.getType(context, entity);
+
+        if (type != null && entityTypes.contains(type.getLabel())) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Checks if the item should be excluded based on the its type
+     *
+     * @param item to be checked
+     * @return whether the item should be processed
+     */
+    protected boolean shouldProcessItemType(Item item) {
+        if (trackerType != null && trackerValues != null) {
+            List<MetadataValue> types = itemService
+                    .getMetadata(item, trackerType.split("\\.")[0], trackerType.split("\\.")[1],
+                                 trackerType.split("\\.").length == 2 ? null : trackerType.split("\\.")[2], Item.ANY);
+
+            if (!types.isEmpty()) {
+                //Find out if we have a type that needs to be excluded
+                for (MetadataValue type : types) {
+                    if (trackerValues.contains(type.getValue().toLowerCase())) {
+                        //We have found no type so process this item
+                        return false;
+                    }
+                }
+                return true;
+            } else {
+                // No types in this item, so not excluded
+                return true;
+            }
+        } else {
+            // No types to be excluded
+            return true;
+        }
+    }
+
+    /**
+     * Initializes services and params obtained from DSpace config
+     */
+    private void initServices() {
```

Comment:
Can we just `Autowired` these services instead of having to init them in a more manual fashion?  This entire method seems like it may not be necessary if we use annotations.

Reply:
Since the processor is not a bean itself, the services could not be autowired. Changed the init to use the factories in declaration.